File: typescript/src/@bubble-sort/bubble-sort.ts
export class BubbleSort {
  sort(array: number[], direction: "asc" | "desc" = "asc"): number[] {
    direction === "desc"
      ? console.log("Ordenando de forma decrescente")
      : console.log("Ordenando de forma crescente");

    for (let i = 0; i < array.length; i++) {
      for (let j = 0; j < array.length; j++) {
        if (direction === "desc") {
          if (array[i] < array[j]) {
            const temp = array[i];
            array[i] = array[j];
            array[j] = temp;
          }
        } else {
          console.log("Ordenando de forma decrescente");
          if (array[i] > array[j]) {
            const temp = array[i];
            array[i] = array[j];
            array[j] = temp;
          }
        }
      }
    }
    return array;
  }
}

File: golang/bubble-sort/bubble-sort.go
package selection_sort

func SelectionSort(list []int, order string) []int {
	if order == "asc" {
		println("Ordenando em ordem crescente")
		for i := 0; i < len(list); i++ {
			for j := 0; j < len(list); j++ {
				if list[i] < list[j] {
					var temp = list[i]
					list[i] = list[j]
					list[j] = temp
				}
			}
		}
		return list
	} else {
		println("Ordenando em ordem decrescente")
		for i := 0; i < len(list); i++ {
			for j := 0; j < len(list); j++ {
				if list[i] > list[j] {
					var temp = list[i]
					list[i] = list[j]
					list[j] = temp
				}
			}
		}
		return list
	}
}
